tests/ops/op_math/test_linear_combination.py

```diff
@@ -0,0 +1,2032 @@
+        H2 = qml.ops.LinearCombination(coeffs2, obs2)
+
+        coeffs_expected = tf.Variable([-4.0, -2.0, -2.0, -1.0])
+        obs_expected = [
+            qml.prod(Y(1), Y(3)),
+            qml.prod(X(0), Y(3)),
+            qml.prod(X(2), Y(1)),
+            qml.prod(X(0), X(2)),
+        ]
+        H = qml.ops.LinearCombination(coeffs_expected, obs_expected)
+
+        assert H.compare(H1 @ H2)
+
+
+class TestLinearCombinationArithmeticTorch:
+    """Tests creation of LinearCombinations using arithmetic
+    operations with torch tensor coefficients."""
+
+    @pytest.mark.torch
+    def test_LinearCombination_equal(self):
+        """Tests equality"""
+        coeffs = torch.tensor([0.5, -1.6])
+        obs = [X(0), Y(1)]
+        H1 = qml.ops.LinearCombination(coeffs, obs)
+
+        coeffs2 = torch.tensor([-1.6, 0.5])
+        obs2 = [Y(1), X(0)]
+        H2 = qml.ops.LinearCombination(coeffs2, obs2)
+
+        assert H1.compare(H2)
+
+    @pytest.mark.torch
+    def test_LinearCombination_add(self):
+        """Tests that LinearCombinations are added correctly"""
+        coeffs = torch.tensor([0.5, -1.6])
+        obs = [X(0), Y(1)]
+        H1 = qml.ops.LinearCombination(coeffs, obs)
+
+        coeffs2 = torch.tensor([0.5, -0.4])
+        H2 = qml.ops.LinearCombination(coeffs2, obs)
+
+        coeffs_expected = torch.tensor([1.0, -2.0])
+        H = qml.ops.LinearCombination(coeffs_expected, obs)
+
+        assert H.compare(H1 + H2)
+
+    @pytest.mark.torch
+    def test_LinearCombination_sub(self):
+        """Tests that LinearCombinations are subtracted correctly"""
+        coeffs = torch.tensor([1.0, -2.0])
+        obs = [X(0), Y(1)]
+        H1 = qml.ops.LinearCombination(coeffs, obs)
+
+        coeffs2 = torch.tensor([0.5, -0.4])
+        H2 = qml.ops.LinearCombination(coeffs2, obs)
+
+        coeffs_expected = torch.tensor([0.5, -1.6])
+        H = qml.ops.LinearCombination(coeffs_expected, obs)
+
+        assert H.compare(H1 - H2)
+
+        H1 -= H2
+        assert H.compare(H1)
+
+    @pytest.mark.torch
+    def test_LinearCombination_matmul(self):
+        """Tests that LinearCombinations are tensored correctly"""
+
+        coeffs = torch.tensor([1.0, 2.0])
+        obs = [X(0), Y(1)]
+        H1 = qml.ops.LinearCombination(coeffs, obs)
+
+        coeffs2 = torch.tensor([-1.0, -2.0])
+        obs2 = [X(2), Y(3)]
+        H2 = qml.ops.LinearCombination(coeffs2, obs2)
+
+        coeffs_expected = torch.tensor([-4.0, -2.0, -2.0, -1.0])
+        obs_expected = [
+            qml.prod(Y(1), Y(3)),
+            qml.prod(X(0), Y(3)),
+            qml.prod(X(2), Y(1)),
+            qml.prod(X(0), X(2)),
+        ]
+        H = qml.ops.LinearCombination(coeffs_expected, obs_expected)
+
+        assert H.compare(H1 @ H2)
+
+
+class TestLinearCombinationArithmeticAutograd:
+    """Tests creation of LinearCombinations using arithmetic
+    operations with autograd tensor coefficients."""
+
+    @pytest.mark.autograd
```

Comment:
You can mark the whole class as `autograd` rather than marking each test. Same for `torch`.

Reply:
✅